.github/workflows/ci.yml

```diff
@@ -17,81 +18,94 @@ on:
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@ba3bb36eb2066252b2363392b7739741bb777659 # v0.8.1
+        with:
+          environments: default lint
+      - name: pre-commit
+        run: pixi run pre-commit-run --color=always --show-diff-on-failure
+
+  tests:
     strategy:
       fail-fast: false
       matrix:
         numfocus_nightly: [false]
         os: ["ubuntu-latest"]
         pandas: [""]
         pyarrow: ["4.0.1", "nightly"]
-        python: ["3.10"]
+        python: ["py311"]
         include:
           - numfocus_nightly: true
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "nightly"
-            python: "3.10"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: "1.5.3"
             pyarrow: "13.0.0"
-            python: "3.11"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "7.0.0"
-            python: "3.10"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "8.0.1"
-            python: "3.10"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "9.0.0"
-            python: "3.10"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "10.0.1"
-            python: "3.11"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "11.0.0"
-            python: "3.11"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "12.0.0"
-            python: "3.11"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "13.0.0"
-            python: "3.11"
+            python: "py311"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "13.0.0"
-            python: "3.12"
+            python: "py312"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "14.0.1"
-            python: "3.12"
+            python: "py312"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "15.0.0"
-            python: "3.12"
+            python: "py312"
           - numfocus_nightly: false
             os: "ubuntu-latest"
             pandas: ""
             pyarrow: "18.1.0"
-            python: "3.12"
+            python: "py312"
```

Comment:
Can you please create specific environments for each CI entry? I.e. this job should be `py312-pyarrow18_1` in the pixi toml.